index.bs

```diff
@@ -3631,6 +3634,377 @@ opened</dfn> steps given |window|, |type| and |message|.
+    dnsStart: uint,
+    dnsEnd: uint,
+    connectStart: uint,
+    connectEnd: uint,
+    tlsStart: uint,
+    tlsEnd: uint,
+    requestStart: uint,
+    responseStart: uint,
+    responseHeadersEnd: uint,
+    responseEnd: uint,
+};
+</pre>
+
+The <code>FetchTimingInfo</code> type represents the time of each part of
+the request, relative to <code>requestTime</code>.
+
+TODO: Add service worker fields
+
+#### The network.Header type #### {#type-network-Header}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+NetworkHeader = {
+    name: text,
+    ? value: text,
+    ? binaryValue: [ uint ]
+};
+</pre>
+
+The <code>Header</code> type represents a single request header.
+
+If the header value can be represented as a UTF-8 encoded string, the
+<code>value</code> field will be present. Otherwise the <code>binaryValue</code>
+field will be present and consist of an array of integers representing the bytes
+of the header.
+
+<div algorithm>
+
+To <dfn>get a Header</dfn> given |name bytes| and |value bytes|:
+
+1. Let |name| be the result of [=UTF-8 decode=] with |name bytes|.
+
+   Assert: Since header names are constrained to be ASCII-only this cannot fail.
+
+1. Let |utf8 decoded value| be the result of [=UTF-8 decode without BOM or fail=]
+   with |value byes|.
+
+1. If |utf8 decoded value| is failure, then:
+
+    1. let |value| be null and |binary value| be a list.
+
+    1. For each |byte| in |value bytes|:
+
+      1. Append the [=value=] of |byte| to |binary value|.
+
+   Otherwise: let |value| be |utf8 decoded value| and let |binary value|
+   be null.
+
+1. Return a map matching the <code>NetworkHeader</code> production, with the
+   <code>name</code> field set to |string name|, the <code>value</code> field
+   set to |value| if it's not null, or omitted otherwise, and the
+   <code>binaryValue</code> field set to |binary value| if it's not null, or
+   omitted otherwise.
+
+#### The network.Initiator type #### {#type-network-Initiator}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+NetworkInitiator = {
+    type: "parser" / "script" / "preflight" / "other",
+    ?columnNumber: uint,
+    ?lineNumber: uint,
+    ?stackTrace: StackTrace,
+    ?request: NetworkRequest
+};
+</pre>
+
+The <code>Initiatior</code> type represents the source of a network request.
+
+TODO: Align more closely with Fetch here?
+
+#### The network.Request type #### {#type-network-RequestId}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl remote-cddl">
+NetworkRequest = text;
+</pre>
+
+Each network request has an associated <dfn export>request id</dfn>, which is a string
+uniquely identifying that request.
+
+#### The network.RequestData type #### {#type-network-RequestData}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+NetworkRequestData = {
+    url: text,
+    method: text,
```

Comment:
Will this be upper case or lower case?

Reply:
It's uppercase, since it matches what's in Fetch.